core/app/models/spree/payment.rb

```diff
@@ -101,7 +111,10 @@ def offsets_total
     # @return [BigDecimal] the amount of this payment minus the offsets
     #   (old-style refunds) and refunds
     def credit_allowed
-      amount - (offsets_total.abs + refunds.sum(:amount))
+      amount - (
+        (self.class.where("source_type = 'Spree::Payment' AND amount < 0 AND state = 'completed' AND source_id = ?", id).sum(:amount)).abs +
```

Comment:
In this case, maybe a variable makes sense? We could add a comment to easily recall which part of the method we need to remove when `offset_payment` will be gone.

Reply:
Given that I needed to repeat the inline on `Spree::Order#has_non_reimbursement_related_refunds?`, and that the code should really be very short-lived (we should remove it in #4989), I'd prefer to keep it simple if you think it's ok :slightly_smiling_face:

Reply:
I had the same suggestion as alberto, but also fine with this give its expected short life